ts/main.ts

```diff
@@ -1,66 +1,326 @@
-if(!$topCategories) throw new Error('.top-cat query failed!')
-const $leftCategories = document.querySelectorAll('.left-cat') as NodeListOf<HTMLDivElement>
-if(!$leftCategories) throw new Error('left-cat query failed!')
-
-function renderGridCategories(gridID: number){
-    for(let i = 0; i < gridSize; i++){
-
-      const imgContainerTop:HTMLDivElement = document.createElement('div')
-      imgContainerTop.classList.add('w-1/2','h-1/2', 'flex', 'justify-center')
-      const catImgTop :HTMLImageElement = document.createElement('img')
-      catImgTop.setAttribute('src', categorySets[gridID].topCategories[i].imgPath)
-      imgContainerTop.appendChild(catImgTop)
-      $topCategories[i].appendChild(imgContainerTop)
-      const catText : HTMLHeadingElement= document.createElement('h1')
-      catText.textContent = categorySets[gridID].topCategories[i].text
-      $topCategories[i].appendChild(catText)
-
-      const imgContainerLeft :HTMLDivElement = document.createElement('div')
-      imgContainerLeft.classList.add('w-1/2', 'h-1/2', 'flex', 'justify-center')
-      const catImgLeft:HTMLImageElement = document.createElement('img')
-      catImgLeft.setAttribute('src', categorySets[gridID].leftCategories[i].imgPath)
-      imgContainerLeft.appendChild(catImgLeft)
-      $leftCategories[i].appendChild(imgContainerLeft)
-      const catTextLeft = document.createElement('h1')
-      catTextLeft.textContent = categorySets[gridID].leftCategories[i].text
-      $leftCategories[i].appendChild(catTextLeft)
+interface Data {
+  archetype: string;
+  atk: number;
+  attribute: string;
+  card_images: ImageURLObj[];
+  card_prices: PricesObj[];
+  card_sets: SetsObj[];
+  desc: string;
+  frameType: string;
+  id: number;
+  linkmarkers: string[];
+  linkval: number;
+  misc_info: MiscInfo[];
+  name: string;
+  race: string;
+  type: string;
+  ygoprodeck_url: string;
+}
+interface Card {
+  data: Data[];
+}
+
+async function fetchData(usrInput: string): Promise<void> {
+  try {
+    const $selectedSq = document.querySelector(
+      '.selected-sq',
+    ) as HTMLDivElement;
+    if (!$selectedSq) throw new Error('.selected-sq query failed!');
+
+    const rowCatNdxStr: string = $selectedSq
+      .getAttribute('id')
+      ?.charAt(4) as string;
+    const rowCatNdx: number = parseInt(rowCatNdxStr);
+    const rowCatCheckFor = categorySets[currentGridId].leftCategories[rowCatNdx]
+      .checkFor as keyof Data;
+    const rowCatVal: number | string =
+      categorySets[currentGridId].leftCategories[rowCatNdx].val;
+
+    const colCatNdxStr: string = $selectedSq
+      .getAttribute('id')
+      ?.charAt(6) as string;
+    const colCatNdx: number = parseInt(colCatNdxStr);
+    const colCatCheckFor = categorySets[currentGridId].topCategories[colCatNdx]
+      .checkFor as keyof Data;
+    const colCatVal: number | string =
+      categorySets[currentGridId].topCategories[colCatNdx].val;
```

Comment:
As a note for later I think we can make this cleaner in a way later, but we can keep this for now.